### DISCUSSION TITLE
Extract head at point rather than cell?

### DISCUSSION THREAD

**Discussion by wkitlasten** [modflowpy/flopy#1445] [General]
I am familiar with getting head values from lay, row, col (and roff, coff) using HOB and getting those from real world coordinates. A few questions:

Is there an arg for layer offset (like roff and coff, but for layers)?

Is there anything that extracts heads from a volume created by interpolating the 3d heads from the binary hds file, or is that considered silly?

**Comment by javgs-bd** [comment]
Hi @wkitlasten, I would suggest to look at the GWUtils of PEST. There are routines for head interpolation from grid nodes (If I recall they are calle usgmod2obs, usgmod2smp, mf6mod2obs and mf6mod2smp).

**Reply by RyanConway91** [reply]
Interpolating within a layer is fine (See GWUtils  or write your own), but interpolating BETWEEN layers is probably not a good idea. Imagine you are interpolating between 2 layers where one is a confining unit and one is a aquifer. Just interpolating a head between these two layer based on distance from the center of layer is not a realistic model of the vertical change in head.

**Reply by wkitlasten** [reply]
@RyanConway91, I can see your point, assuming you "know" the location of confining layers and aquifer materials well enough to define your model structure/layering and you "know" where your observation points fall relative to those layers. However, in larger regional models the layering rarely lines up with geology (e.g., 5m confining units within 20m thick layers), so approximations must be made. For example, if it just so happens that one observation falls at the top of a cell/layer with low Kz and another falls just at the bottom of the same cell/layer, does it make sense they should be considered to have the same simulated value?
